File: fixtures/scale/python/large/cipherlab/checks/check_cascadestream.py
from cipherlab.ciphers.cascadestream import cascadestream_encrypt, cascadestream_decrypt
from cipherlab.specs.cascadestream_spec import cascadestream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_cascadestream():
    spec = cascadestream_spec()
    for text in corpus_stream():
        sealed = cascadestream_encrypt(text, spec["key"])
        opened = cascadestream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"

File: fixtures/scale/ruby/large/lib/check_cascadestream.rb
require_relative 'cascadestream_encrypt'
require_relative 'cascadestream_decrypt'
require_relative 'corpus_stream'

def check_cascadestream
  corpus_stream.each do |text|
    sealed = cascadestream_encrypt(text, 11)
    opened = cascadestream_decrypt(sealed, 11)
    return false if opened != text
  end
  true
end
